nextcord/http.py

```diff
@@ -182,49 +440,144 @@ def __exit__(
-        self.connector = connector
+        # TODO: Think about adding ratelimit_multiplier? Would reduce the internal RateLimit.limit by that
+        #  float (such as 0.7) and could allow people to run multiple NC bots/processes on the same token while avoiding
+        #  ratelimit issues. Could also help with replit-style scenarios.
         self.__session: aiohttp.ClientSession = MISSING  # filled in static_login
-        self._locks: weakref.WeakValueDictionary = weakref.WeakValueDictionary()
-        self._global_over: asyncio.Event = asyncio.Event()
-        self._global_over.set()
-        self.token: Optional[str] = None
-        self.bot_token: bool = False
-        self.proxy: Optional[str] = proxy
-        self.proxy_auth: Optional[aiohttp.BasicAuth] = proxy_auth
-        self.use_clock: bool = not unsync_clock
-        self._dispatch: Callable = dispatch
+        self._connector = connector
+        self._default_max_per_second = default_max_per_second
+        """Maximum amount of requests per second per authorization."""
+        self._time_offset = time_offset
+        """Amount of seconds added to all ratelimit timers for lag compensation."""
+        self._default_auth = None
+        # For consistency with possible future changes to set_default_auth.
+        self._set_default_auth(default_auth)
+        self._proxy = proxy
+        self._proxy_auth: Optional[aiohttp.BasicAuth] = proxy_auth
+        # loop is truthy by default it seems, so this works.
+        self._loop: asyncio.AbstractEventLoop = loop or asyncio.get_event_loop()
+        self._dispatch = dispatch
 
         user_agent = "DiscordBot (https://github.com/nextcord/nextcord/ {0}) Python/{1[0]}.{1[1]} aiohttp/{2}"
-        self.user_agent: str = user_agent.format(__version__, sys.version_info, aiohttp.__version__)
+        self._user_agent: str = user_agent.format(
+            __version__, sys.version_info, aiohttp.__version__
+        )
+
+        self._buckets: dict[str, RateLimit] = {}
+        """{"Discord bucket name": RateLimit}"""
+        self._global_rate_limits: dict[str | None, RateLimit] = {}
+        """{"Auth string": RateLimit}, None for auth-less ratelimit."""
+        self._url_rate_limits: dict[tuple[str, str, str | None], RateLimit] = {}
+        """{("METHOD", "Route.bucket", "auth string"): RateLimit} auth string may be None to indicate auth-less."""
+        # TODO: Is this worth keeping? May reduce errors, thus cloudflare bannability.
+        self._url_deny_list: set[tuple[str, str, str | None]] = set()
+        """{("METHOD", "route", "auth string"), ...}
+
+        Paths that result in a 404. These will immediately error if a request is made to them.
+        """
+
+    def _make_global_rate_limit(self, auth: str | None, max_per_second: int) -> GlobalRateLimit:
+        _log.info(
+            "Creating global ratelimit for auth %s with max per second %s.", auth, max_per_second
+        )
```

Comment:
Yea, no. I don't approve of bot tokens being in logs, especially at info level

Reply:
Fair, the latest commit should be an adequate fix?